tests/protocol/test_wire.py

```diff
@@ -838,50 +835,18 @@ def http_get_handler(url, *_, **kwargs):
             self.assertTrue(self.is_host_plugin_extension_artifact_request(urls[3]), "The retry request should have been over the host channel")
             self.assertFalse(HostPluginProtocol.is_default_channel, "The default channel should not have changed")
 
-    def test_upload_logs_should_not_refresh_plugin_when_first_attempt_succeeds(self):
-        def http_put_handler(url, *_, **__):  # pylint: disable=inconsistent-return-statements
-            if self.is_host_plugin_put_logs_request(url):
-                return MockHttpResponse(200)
-
-        with mock_wire_protocol(mockwiredata.DATA_FILE, http_put_handler=http_put_handler) as protocol:
-            content = b"test"
-            protocol.client.upload_logs(content)
-
-            urls = protocol.get_tracked_urls()
-            self.assertEqual(len(urls), 1, 'Expected one post request to the host: [{0}]'.format(urls))
-
-    def test_upload_logs_should_retry_the_host_channel_after_refreshing_the_host_plugin(self):
-        def http_put_handler(url, *_, **__):
-            if self.is_host_plugin_put_logs_request(url):
-                if http_put_handler.host_plugin_calls == 0:
-                    http_put_handler.host_plugin_calls += 1
-                    return ResourceGoneError("Exception to fake a stale goal state")
-                protocol.track_url(url)
-            return None
-        http_put_handler.host_plugin_calls = 0
-
-        with mock_wire_protocol(mockwiredata.DATA_FILE_IN_VM_ARTIFACTS_PROFILE, http_put_handler=http_put_handler) \
-                as protocol:
-            content = b"test"
-            protocol.client.upload_logs(content)
-
-            urls = protocol.get_tracked_urls()
-            self.assertEqual(len(urls), 2, "Invalid number of requests: [{0}]".format(urls))
-            self.assertTrue(self.is_host_plugin_put_logs_request(urls[0]), "The first request should have been over the host channel")
-            self.assertTrue(self.is_host_plugin_put_logs_request(urls[1]), "The second request should have been over the host channel")
-
     @staticmethod
     def _set_and_fail_helper_channel_functions(fail_direct=False, fail_host=False):
         def direct_func(*_):
             direct_func.counter += 1
             if direct_func.fail:
-                return None
+                raise Exception("Direct channel failed")
```

Comment:
The mocks need to be updated to raise exceptions instead of reporting errors with None or False